doc/devguide/extending/app-layer/transactions.rst

```diff
@@ -0,0 +1,299 @@
+************
+Transactions
+************
+
+.. contents:: Table of Contents
+
+_`General Concepts`
+===================
+
+Transactions are abstractions that help detecting and logging in Suricata. They also help during the detection phase,
+when dealing with protocols that can have large PDUs, like TCP, in controlling state for partial rule matching, in case of rules that mention more than one field.
```

Comment:
This does not seem clear to me as an introduction. I would rather begin with an example like a HTTP request and its response constitute one transaction